cli/src/main.rs

```diff
@@ -246,8 +252,28 @@ fn compile_once(world: &mut SystemWorld, command: &CompileSettings) -> StrResult
     match typst::compile(world) {
         // Export the PDF.
         Ok(document) => {
-            let buffer = typst::export::pdf(&document);
-            fs::write(&command.output, buffer).map_err(|_| "failed to write PDF file")?;
+            if command.output.extension() == Some(OsStr::new("png")) {
```

Comment:
Currently this would not recognize a different capitalization like `PNG`. Maybe adding a `to_lower` might be appropriate.

Reply:
This would involve a lot of somewhat gross unwrapping (because it's an OsStr), are you aware of a better way of doing this check, this feels rather gross.

Reply:
Perhaps something like

```rust
if command.output.extension().map_or(false, |ext| ext.eq_ignore_ascii_case("png")) {
   // ...
}
```

Reply:
This looks good to me. But as in the future more extensions might be added, a cleaner (although much more time intensive) solution would be to have an `Extension` enum that parses the file extension and then gets matched on.

Reply:
Perhaps something for another PR, for now I'll go with @PgBiel's solution. I was looking into using a match statement but that wouldn't solve the problem of lowercasing unfortunately.

Reply:
> I was looking into using a match statement but that wouldn't solve the problem of lowercasing unfortunately.
A match with a guard could also work:
```rust
match command.output.extension() {
  Some(ext) if ext.eq_ignore_ascii_case("png") => { /* if branch */ }
  _ => { /* else branch */ }
};
```
I wonder if that's clearer than the current approach or not, but it's certainly possible that way